lib/edition_filter.rb

```diff
@@ -90,4 +91,11 @@ def ordered_scope(scope)
       scope
     end
   end
+
+  def pre_release_document_type_scope(scope)
+    return scope if user.has_permission?(User::PRE_RELEASE_FEATURES_PERMISSION)
+
+    ids = scope.map { |edition| edition.id unless edition.document_type.pre_release? }.compact
```

Comment:
This is going to be inefficient because it involves loading all of the documents into memory, and then paginating them to get back to a smaller set. We want to preserve the original property of this code where all the processing is done in the DB, which avoids shipping loads of redundant data back to the app and keeps page load times low.

We should be able to do this by figuring out allowed set of document type IDs (ones that aren't pre-release) and doing a `where` query on those.

Reply:
Thanks for your help with this, was struggling with it yesterday!